feat(apple-silicon): add private networks feature (#4138)
scaleway/docs-content: bare-metal/apple-silicon/how-to/use-private-networks.mdx

tags: mac-mini mac-mini apple-silicon private-network vpc
dates:
  validation: 2024-12-19
  posted: 2021-05-26
categories:
  - bare-metal
---

Private Networks allow your Mac mini to communicate in an isolated and secure network with other Scaleway resources, away from the public internet. This is an optional-billed feature for Mac mini.

Each Mac mini can be connected to one or several Private Networks, letting you build your own network topologies. Private Networks’ built-in DHCP server manages the allocation of private IP addresses for Mac mini servers (and other resources) when they are attached to the network.

For full information about Scaleway Private Networks and VPC, see our [dedicated documentation](/network/vpc/)

<Macro id="requirements" />

- A Scaleway account logged into the [console](https://console.scaleway.com)
- [Owner](/identity-and-access-management/iam/concepts/#owner) status or [IAM permissions](/identity-and-access-management/iam/concepts/#permission) allowing you to perform actions in the intended Organization
- A [Mac mini](/bare-metal/apple-silicon/how-to/create-mac-mini/)

## How to enable the Private Networks feature

> Before you can attach your Mac mini to a Private Network, you must enable the feature. This can be done during [Mac mini creation](/bare-metal/apple-silicon/how-to/create-mac-mini/) or afterwards. These instructions show how to enable the feature after creation of the Mac mini.

Reviewer: ```suggestion
Before you can attach your Mac mini to a Private Network, you must enable the feature. This can be done during [Mac mini creation](/bare-metal/apple-silicon/how-to/create-mac-mini/) or afterwards. These instructions show how to enable the feature after the creation of the Mac mini.
```

4. Click **Enable Private Networks**.

Private Networks are enabled, and you are returned to your Mac mini's **Overview** page.

## How to disable the Private Networks feature

You can disable the Private Networks feature at any time after activating it.

1. Click **Apple silicon** in the **Bare Metal** section of the side menu. The Apple silicon splash page displays.
2. Click the Mac mini you want to disable Private Networks on. The Mac mini's **Overview** page displays.
3. In the **Private Networks feature** panel, use the <Icon name="toggle" /> toggle button to disable the feature.
    A pop-up displays, asking you to confirm that you want to disable Private Networks.
4. Click **Disable Private Networks**.

Private Networks are disabled, and you are returned to your Mac mini's **Overview** page. Any Private Networks previously attached to the Mac mini are now detached. 

## How to attach a Mac mini to a Private Network

1. Click **Apple silicon** in the **Bare Metal** section of the side menu. The Apple silicon splash page displays.
2. Click the Mac mini you want to attach to a Private Network. The Mac mini's **Overview** page displays.

> 3. Click the **Private Networks**tab.

Reviewer: ```suggestion
3. Click the **Private Networks** tab.
```

4. Click **+ Attach to a Private Network**. A pop-up displays.

> 5. Select the Private Network you want to attach the Mac mini to.  It will be automatically assigned private IPv4 and IPv6 addresses on this network. The ability to use specific reserved IP addresses for the attachment is planned for the future.

Reviewer: ```suggestion
5. Select the Private Network you want to attach the Mac mini to. It will be automatically assigned private IPv4 and IPv6 addresses on this network. The ability to use specific reserved IP addresses for the attachment is planned for the future.
```

6. Click **Attach to Private Network**.

The Mac mini is attached to the Private Network, and you are returned to the **Private Networks** tab. The IPv4 and the IPv6 addresses for the Mac mini on the Private Network are displayed, along with the ID of the VLAN interface that was created.

## How to detach a Mac mini from a Private Network

1. Click **Apple silicon** in the **Bare Metal** section of the side menu. The Apple silicon splash page displays.
2. Click the Mac mini you want to detach from a Private Network. The Mac mini's **Overview** page displays.

> 3. Click the **Private Networks**tab.

Reviewer: ```suggestion
3. Click the **Private Networks** tab.
```

4. Click the unlink icon <Icon name="unlink" /> next to the Private Network you want to detach from the Mac mini.

>     A pop-up displays, asking you to confirmt hat you want to detach the resource.

Reviewer: ```suggestion
    A pop-up displays, asking you to confirm that you want to detach the resource.
```

The Mac mini is attached to the Private Network, and you are returned to the **Private Networks** tab. The IPv4 and the IPv6 addresses for the Mac mini on the Private Network are displayed, along with the ID of the VLAN interface that was created.

Once you have attached the resource, some manual configuration of the network interface is required. See the next section for full details.

## How to configure the VLAN interface on your Mac mini for Private Networks

> You must configure the virtual network interface on each Mac mini you have attached to a Private Network. Traffic from the public internet and traffic from a Private Network will pass through the same network interface of the Mac mini. We must therefore separate the traffic by adding a VLAN interface.

Reviewer 2: ```suggestion
You must configure the virtual network interface on each Mac mini you have attached to a Private Network. Traffic from the public internet and traffic from a Private Network will pass through the same physical network interface of the Mac mini. We must therefore separate the traffic by adding a VLAN interface.
Comment
